wg-api/spec-documents/scroll-view.md

```diff
@@ -0,0 +1,259 @@
+# Scroll View
+
+## Summary
+
+This spec proposes to add an experimental native ScrollView component to Electron.
+
+## Platforms
+* macOS 
+* Windows
+* Linux
+
+## Impetus
+
+It is currently not possible to add scrolling functionality around BrowserViews:
+
+* Capture mouse / touch events
+* Reposition BrowserViews accordingly
+
+More details of the use-case and challenges related to workarounds: [#32751](https://github.com/electron/electron/issues/32751)
+
+![](https://user-images.githubusercontent.com/2891206/150981006-84fe796e-60bd-43c0-952d-831d1bfd9998.gif)
+
+## Usage Example
+
+```js
+const { BrowserView, BrowserWindow, ContainerView, ScrollView, WrapperBrowserView } = require("electron");
+
+const window = new BrowserWindow({ width: 1000, height: 500 })
+
+const scroll = new ScrollView()
+scroll.setBounds({ width: 1000, height: 500 })
+
+const scrollContent = new ContainerView()
+scrollContent.setBounds({ width: 2000, height: 500 })
+
+const browserView1 = new BrowserView()
+const wrapperBrowserView1 = new WrapperBrowserView({ 'browserView': browserView1 });
+wrapperBrowserView1.setBounds({ width: 1000, height: 500 })
+scrollContent.addBrowserView(wrapperBrowserView1)
+
+const browserView2 = new BrowserView()
+const wrapperBrowserView2 = new WrapperBrowserView({ 'browserView': browserView2 });
+wrapperBrowserView2.setBounds({ width: 1000, height: 500, x: 1000 })
+scrollContent.addBrowserView(wrapperBrowserView2)
+
+scroll.setContentView(scrollContent);
+win.addChildView(scroll);
+```
+
+
+## API Design
+
+In order to achieve the example above, we've made following changes to the API
+
+## BaseView
+
+A `BaseView` is a rectangle within the views View hierarchy. It is the base
+class for all all different views. 
+
+Process: **Main**
+
+`BaseView` is an EventEmitter
+
+### Static Methods
+
+#### `BaseView.getAllViews()`
+
+Returns the array of all created views.
+
+#### `BaseView.fromId(id)`
+
+Returns `BaseView | null` - The view with the given `id`.
+
+### Instance Properties
+
+#### `view.id`
+
+A `Integer` property representing the unique ID of the view.
+
+#### `view.isContainer`
+
+A `boolean` property that determines whether this view is `ContainerView`.
+
+### Instance Methods
+
+#### `view.setBounds(bounds)`
+
+Resizes and moves the view to the supplied bounds relative to its parent.
+
+#### `view.getBounds()`
+
+Returns the position and size of the view.
+
+#### `view.setBackgroundColor(color)`
+
+Change the background color of the view.
+
+#### `view.getParentView()`
+
+Returns `BaseView || null` - The parent view, otherwise returns `null`.
+
+#### `view.getParentWindow()`
```

Comment:
nit: I don't think `parent` is the best name here; that implies a direct relationship, but there are possibly several views in the hierarchy between the view and the window. Perhaps `getOwnerWindow` or just `getWindow`?

Reply:
That's the relationship; thus a view may be a child of either another view or a window. The best would be to have baseView.getParent() returning BaseView|BaseWindow.

Reply:
hmm, i disagree. a Window seems like a categorically different thing from a View, it would be weird to have one function to return either. the relationship is "ancestor" not "parent", no?

Reply:
That's parent relationship :)

Reply:
I'm confused.

```
window
+-- view1
    +-- view2
```

`window` is not the parent of `view2`. I would argue that calling it the parent of `view1` is also inaccurate, because conflating the _window_ with the _view tree_ is not a good metaphor. The window has a single root view that has no parent, which is the top of the view tree.

calling `view2.getWindow()` should return `window`.

Reply:
Well, I agree, that should looks like that.
For the following scenario:
```
browserwindow1.addChild(view1)
containerView1.addChild(view2)
```
- view1.getParentView() will return null
- view2.getParentView() will return containerView1
I assumed, that
- view2.getParentWindow() will return null
- view1.getParentWindow() will return browserWindow1
in this way, it would be possible to distinguish to which object a view has been added.
We can replace getParentWindow by getWindow - then both view1.getWindow () and view2.getWindow () would return browserWindow1 - no prob.